### ISSUE TITLE
Clarify Boost 1.54 dependency caveat in documentation

### ISSUE THREAD

**Issue by anonimal** [monero-project/monero#906]
```
anonimal        Is the boost 1.54 caveat still applicable?
anonimal        I've built successfully with gcc 4.8.5 and 5.3.0 on ubuntu LTS with 1.54, and both nodes are syncing successfully.
anonimal        Does the error/issue come up later?
i2p-relay       {-fluffypony} anonimal: it's Windows-specific, I think
```

I'm writing up a Travi-CI yaml for bitmonero so a solid answer will help me build a more accurate recipe.

**Comment by NanoAkron** [comment]
I think that according to #1015, we should now be using Boost 1.58

**Comment by anonimal** [comment]
Resolved in #964
